### PR TITLE
Fix conformance tests for Proposal-4

### PR DESCRIPTION
**Pull request by danylo-lebediev-capgemini**
This fix adds generation of valid wight/height of image for Proposal-4.

### CODE REVIEW

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp [hunk 1]
```diff
@@ -548,6 +548,16 @@ ov::runtime::Tensor generate(const std::shared_ptr<ngraph::op::v4::Proposal>& no
                              const ov::Shape& targetShape) {
     if (port == 1) {
         return ov::test::utils::create_and_fill_tensor_normal_distribution(elemType, targetShape, 0.0f, 0.2f, 7235346);
+    } else if (port == 2) {
+        ov::Tensor tensor = ov::Tensor(elemType, targetShape);
+
+        auto *dataPtr = tensor.data<float>();
+        dataPtr[0] = dataPtr[1] = 225.0f;
+        dataPtr[2] = 1.0f;
+        if (tensor.get_size() == 4)
```
Comment (ceciliapeng2011) [inline src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:557]: May I ask the root cause of this conformance test failure? Why updating the inputs this way could be a fix? Thanks.

Comment (iefode) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:557]: @ceciliapeng2011 
Some input generation for some operations should be specialized. Proposal is part of it. 
Refer to specification input[2] should be picture size. But as for Me better to use other data type: `tensor.data<size_t>` for example

Comment (danylo-lebediev-capgemini) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:557]: According to the specification https://docs.openvino.ai/latest/openvino_docs_ops_detection_Proposal_4.html, the input[2] can only be of floating-point type, so we can't change the type here.

**Review by ceciliapeng2011 — APPROVED** [review]
LGTM

**Comment by lohika-denis-kotov** [comment]
@ceciliapeng2011 @iefode By some reason ci/jenkins stuck for long time and Auto-Merge could not be accomplished

**Comment by iefode** [comment]
> @ceciliapeng2011 @iefode By some reason ci/jenkins stuck for long time and Auto-Merge could not be accomplished
I will start CI soon

**Review by iefode — APPROVED** [review]

**Pull request merged**